### PR TITLE
Refactor ApplicationChoiceComponent

### PR DESCRIPTION
**Pull request by george-bit**
## Context

Tech debt

## Changes proposed in this pull request

This PR refactors the rows method on this view component to draw from several private methods for greater efficiency and
readability.

## Guidance to review

Could this be refactored any further?

## Link to Trello card

https://trello.com/c/9FLsTXEg/3362-refactor-the-supportinterfaceapplicationchoicecomponent

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/support_interface/application_choice_component.rb [hunk 1]
```diff
@@ -10,92 +10,139 @@ def initialize(application_choice)
...
-      rows << { key: 'Sent to provider at', value: application_choice.sent_to_provider_at.to_s(:govuk_date_and_time) } if application_choice.sent_to_provider_at
-      rows << { key: 'Reject by default at', value: application_choice.reject_by_default_at.to_s(:govuk_date_and_time) } if application_choice.reject_by_default_at && application_choice.decision_pending?
-      rows << { key: 'Decline by default at', value: application_choice.decline_by_default_at.to_s(:govuk_date_and_time) } if application_choice.decline_by_default_at && application_choice.offer?
+    def rejection_reason_row
+      return unless rejection_reasons_text
 
+      { key: 'Rejection reason', value: rejection_reasons_text }
+    end
+
+    def sent_to_provider_at_row
```
Comment (davidgisbey) [inline app/components/support_interface/application_choice_component.rb:100]: super nitty but 

`return unless application_choice.sent_to_provider_at`

**Review by davidgisbey — APPROVED** [review]
Looks good to me George. This is way easier to parse now

File: app/components/support_interface/application_choice_component.rb [hunk 2]
```diff
@@ -10,92 +10,139 @@ def initialize(application_choice)
...
+      { key: 'Sent to provider at', value: application_choice.sent_to_provider_at.to_s(:govuk_date_and_time) } if application_choice.sent_to_provider_at
+    end
+
+    def reject_by_default_at_row
```
Comment (davidgisbey) [inline app/components/support_interface/application_choice_component.rb:104]: I think this is covered on line 84 right?

Comment (george-bit) [reply app/components/support_interface/application_choice_component.rb:104]: Not quite... as I understand `reject_by_default_at` is a date in the future that is generated when app choice is decision pending.

Line 84 displays `rejected_at` which is a date generic to any rejected application choice. The logic around this draws from `rejected_by_default` which is a boolean set to true when an application choice is rejected by default.

TL;DR - This is present/future tense theother is past tense and they pull dates from different db columns.

Comment (davidgisbey) [reply app/components/support_interface/application_choice_component.rb:104]: Right, sorry about that i missed the tense 👍  Good to go then

**Review by JR-G — APPROVED** [review]
Nice one George! Much cleaner 🙌
